virttest/utils_test/__init__.py

```diff
@@ -1947,149 +1959,143 @@ def _tarball_download(self, url, destination):
-        session.close()
-
+        self.cmd_output_safe(rm_cmd)
 
-class HostStress(object):
 
+class VMStress(Stress):
     """
-    Run Stress tool on host, such as stress, unixbench, iozone and etc.
+    Run Stress tool on VMs, such as stress, unixbench, iozone and etc.
     """
-
-    def __init__(self, params, stress_type):
-        """
-        Set parameters for stress type
+    def __init__(self, vm, stress_type, params, download_url="", make_cmds="",
+                 stress_cmds="", stress_args="", work_path="",
+                 uninstall_cmds="", download_type="tarball"):
         """
+        Set parameters for stress type, for the arguments have default value "",
+        they can be either passed here, or defined in vm.params
 
-        def _parameters_filter(stress_type):
-            """Set parameters according stress_type"""
-            _control_files = {'unixbench': "unixbench5.control",
-                              'stress': "stress.control"}
-            _check_cmds = {'unixbench': "pidof -s ./Run",
-                           'stress': "pidof -s stress"}
-            _stop_cmds = {'unixbench': "killall ./Run",
-                          'stress': "killall stress"}
-            try:
-                control_file = _control_files[stress_type]
-                self.control_path = os.path.join(data_dir.get_root_dir(),
-                                                 "shared/control",
-                                                 control_file)
-                self.check_cmd = _check_cmds[stress_type]
-                self.stop_cmd = _stop_cmds[stress_type]
-            except KeyError:
-                self.control_path = ""
-                self.check_cmd = ""
-                self.stop_cmd = ""
-
-        self.params = {}
-        if params:
-            self.params = params
-        self.timeout = 60
-        self.stress_type = stress_type
-        self.host_stress_process = None
-        if stress_type not in ["stress", "unixbench"]:
-            raise StressError("Stress %s is not supported now." % stress_type)
-
-        _parameters_filter(stress_type)
-        self.stress_args = self.params.get("stress_args", "")
+        :param vm: the vm to be loading stress
+        :param stress_type: the name of the stress tool
+        :param params: Test dict params
+        :param download_url: from where download the stress tool
+        :param make_cmds: make command of the stress tool
+        :param uninstall_cmds: uninstall command of the stress tool
+        :param work_path: the relative work path of the stress tool,
+        e.g. for iozone: "src/current"
+        :param stress_cmds: the command to launch stress,
+        use stress_type instead if not defined
+        :param stress_args: the arguments of the stress tool
+        :param download_type: currently support "git" or "tarball"
+        """
+        super(VMStress, self).__init__(stress_type, params, download_url="",
```

Comment:
@balamuruhans, i find an issue here. When we call the parent class here, we shouldn't give its params a default value, like "download_url = ‘’"
 If so,  when i call VMStress with "download_url = "https://www.xxx.iozone"", it will finally get download_url = "".